app/drivers/sensor/max17048/max17048.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT maxim_max17048
+
+#include <device.h>
+#include <kernel.h>
+#include <sys/util.h>
+#include <logging/log.h>
+#include <drivers/i2c.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+
+#include "max17048.h"
+
+LOG_MODULE_REGISTER(MAX17048, CONFIG_SENSOR_LOG_LEVEL);
+
+static int read_register(const struct device *dev, uint8_t reg, uint16_t *value) {
+
+    if (k_is_in_isr()) {
+        return -EWOULDBLOCK;
+    }
+
+    struct max17048_drv_data *const drv_data = dev->data;
+    uint16_t dev_addr = ((struct max17048_config *)dev->config)->device_addr;
+
+    uint16_t data = 0;
```

Comment:
some people on the discord have noted that this code doesn't work very well, and should be changed to `uint8_t[2]` instead. will fix soon.